packages/auth-construct/src/construct.ts

```diff
@@ -282,6 +253,48 @@ export class AmplifyAuth
     return result;
   };
 
+  /**
+   * Auto generate the user pool groups and group roles
+   */
+  private setupUserPoolGroups = (
+    groups: string[] | undefined,
+    identityPool: CfnIdentityPool
+  ) => {
+    if (groups) {
```

Comment:
nit: prefer early return on `if (!groups)`

Alternatively, we could replace the line below with `(groups || []).forEach(...)` and remove this if entirely

Reply:
updated